### ISSUE TITLE
wkhtmltopdf 0.12.1 does not work on Ubuntu 18.04 because of dependency issue

### ISSUE THREAD

**Issue by Yenthe666** [wkhtmltopdf/wkhtmltopdf#3898]
Hi there,

When you download and try to install wkhtmltopdf you'll see that Ubuntu 18.04 does not support `libpng12-0` (also see https://packages.ubuntu.com/search?keywords=libpng12-0) when you run the following commands:
```
sudo wget https://downloads.wkhtmltopdf.org/0.12/0.12.1/wkhtmltox-0.12.1_linux-trusty-amd64.deb
sudo gdebi --n `basename https://downloads.wkhtmltopdf.org/0.12/0.12.1/wkhtmltox-0.12.1_linux-trusty-amd64.deb`
```

Output:
```
Reading package lists... Done
Building dependency tree        
Reading state information... Done
Reading state information... Done
This package is not installable
Dependency is not satisfiable: libpng12-0

```

**Comment by ashkulz** [comment]
Ubuntu 18.04 got released 4 days ago, and you're trying to install `0.12.1` which is 3 releases old :frowning:

You can contribute a PR for supporting Bionic in [wkhtmltopdf/packaging](https://github.com/wkhtmltopdf/packaging) or see if the stretch / xenial builds work for you.

**Comment by Yenthe666** [comment]
Hi @ashkulz,

I know this is quite a lot behind. Sadly Odoo itself only has a full support on `0.12.1`, which forces us to stick to it (see https://github.com/odoo/odoo/wiki/Wkhtmltopdf)

**Comment by ashkulz** [comment]
I'd [love to know what issues](https://github.com/odoo/odoo/issues/21255#issuecomment-385242505) Odoo is facing with the latest builds, as I'd like fix them before the 0.12.5 release.

**Comment by ashkulz** [comment]
You got your wish via wkhtmltopdf/packaging@52b4e170592c741ea661e582a23c6a8b9b697893 -- is there any OS/distro/arch combination that you'd like to see packaged as well?

**Comment by Yenthe666** [comment]
@ashkulz nice! No, this is the only OS/distro that I need.
Just wondering, can this then be applied on wkhtmltopdf `0.12.1` so that if I download it from `https://downloads.wkhtmltopdf.org/0.12/0.12.1/wkhtmltox-0.12.1_linux-trusty-amd64.deb` it'll work or not? Wondering how to deal correctly with it in an older version.

**Comment by ashkulz** [comment]
See https://builds.wkhtmltopdf.org/0.12.1.3/ -- I had posted the link in the other ticket.

**Comment by Yenthe666** [comment]
@ashkulz on the page https://github.com/wkhtmltopdf/wkhtmltopdf/releases/0.12.3 I'm missing a `wkhtmltox-0.12.3_linux-trusty-amd64.deb` equivalt of the one at page https://github.com/wkhtmltopdf/wkhtmltopdf/releases/0.12.1 (named `wkhtmltox-0.12.1_linux-trusty-amd64.deb`).
The 0.12.3 page has no .deb packages?

**Comment by ashkulz** [comment]
Not really getting what you are asking. The `0.12.3` build switched to a generic build based on CentOS 6, but it obviously doesn't work everywhere so switching back to distro-specific builds. I've created a special release **0.12.1**.3 for odoo before they can confirm what's missing in `0.12.5`, which was what you wanted...

**Comment by treviser** [comment]
@Yenthe666 I did succesfully install and test the new build from @ashkulz , see https://www.odoo.com/forum/help-1/question/ubuntu-18-04-lts-how-to-install-wkhtmltopdf-0-12-1-134198

**Comment by jcarlosmontoya** [comment]
Hello All, 
Im trying to print pdf in Odoo 12 on Ubuntu 18.04.1 LTS Desktop 64bits and it show this error
odoo.exceptions.UserError: ("Wkhtmltopdf failed (error code: -8). Mensaje: b''", '')
I have tested with Wkhtmltopdf 0.12.1.3, 0.12.5 latest

**Comment by JKE-be** [comment]
@jcarlosmontoya 
body {
  font-family: ..., "Noto Color Emoji";
}
from boostrap 4

You can install ttf-mscorefonts-installer to fix it.

@ashkulz https://bugs.launchpad.net/ubuntu/+source/wkhtmltopdf/+bug/1363650

**Comment by odony** [comment]
@jcarlosmontoya @JKE-be let's track this specific issue separately: #4149

**Comment by sergueikp** [comment]
> @ashkulz yeah don't worry this is enough for now! Give the Odoo team some time to test and get back to you, we'll take it from there! Thanks a lot for the help / new release.

Hello Yenthee, 
I experience a problem with wkhtmltox_0.12.1.4-2  with my old Odoo 8.0. Recently I moved it to Ubuntu 18.04 server, 
everything works well untill I install wkhtmltopdf  package, server stops to respond, but if I remove the package
server works fine again. 
I followed those instructions to install the package:
sudo wget https://github.com/wkhtmltopdf/packaging/releases/download/0.12.1.4-2/wkhtmltox_0.12.1.4-2.bionic_amd64.deb
sudo apt install -f ./wkhtmltox_0.12.1.4-2.bionic_amd64.deb
sudo ln -s /usr/local/bin/wkhtmltopdf /usr/bin
sudo ln -s /usr/local/bin/wkhtmltoimage /usr/bin

I see in openrp-server.log this error:
version = re.search('([0-9.]+)', out).group(0)
AttributeError: 'NoneType' object has no attribute 'group'

This means that odoo can not get the version of the wkhtmltopdf package from the system!
Question is how to provide the version info to Odoo?

But if I uninstall the wkhtmltopdf package server start to work fine again.
Thank you!
